chapters/what_neural_search.md

```diff
@@ -0,0 +1,103 @@
+# What is Neural Search?
```

Comment:
Just one thing, this should be in title case

Reply:
Fixed!